### PR TITLE
Fix change in get authenticated user name method

### PR DESCRIPTION
**Pull request by sadilchamishka**

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java [hunk 1]
```diff
@@ -188,12 +191,17 @@ public static String getOrganizationId() {
...
         if (username == null) {
-            return organizationUserResidentResolverService.resolveUserFromResidentOrganization(null,
-                    getUserId(), getOrganizationId()).map(User::getUsername).orElse(null);
+            try {
+                username = organizationUserResidentResolverService.resolveUserFromResidentOrganization(null,
+                        getUserId(), getOrganizationId()).map(User::getUsername).orElse(null);
+            } catch (OrganizationManagementException e) {
+                LOG.debug("Authenticated user name could not resolved.", e);
+            }
+
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java:204]: remove additional new line

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java [hunk 2]
```diff
@@ -188,12 +191,17 @@ public static String getOrganizationId() {
...
 
         String username = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
         if (username == null) {
-            return organizationUserResidentResolverService.resolveUserFromResidentOrganization(null,
-                    getUserId(), getOrganizationId()).map(User::getUsername).orElse(null);
+            try {
+                username = organizationUserResidentResolverService.resolveUserFromResidentOrganization(null,
+                        getUserId(), getOrganizationId()).map(User::getUsername).orElse(null);
+            } catch (OrganizationManagementException e) {
+                LOG.debug("Authenticated user name could not resolved.", e);
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java:202]:
```suggestion
                LOG.debug("Authenticated username could not be resolved.", e);
```

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java (hunk 1 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java]: Fixed by - https://github.com/wso2-extensions/identity-organization-management-core/pull/12/commits/1edee38bbd81881834fcd790409c7a8c7eaec86f

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java (hunk 2 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java]: Fixed by - https://github.com/wso2-extensions/identity-organization-management-core/pull/12/commits/1edee38bbd81881834fcd790409c7a8c7eaec86f

**Review by AnuradhaSK — APPROVED** [review]

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java [hunk 3]
```diff
@@ -188,12 +191,16 @@ public static String getOrganizationId() {
...
 
         String username = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
         if (username == null) {
-            return organizationUserResidentResolverService.resolveUserFromResidentOrganization(null,
-                    getUserId(), getOrganizationId()).map(User::getUsername).orElse(null);
+            try {
+                username = organizationUserResidentResolverService.resolveUserFromResidentOrganization(null,
+                        getUserId(), getOrganizationId()).map(User::getUsername).orElse(null);
+            } catch (OrganizationManagementException e) {
+                LOG.debug("Authenticated user's username could not be resolved.", e);
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java:202]: what's the reason for this change?

Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java:202]: Previously when organization is created, a tenant is created behind the scene with the organization creator's ID is set to the tenant admin's username. Later it was changed to keep the organization creator's username as the tenant admin's username. 

The authenticated organization creator can resides in higher level organization, but accessing a lower level organization to create an organization. This should be a cross tenant flow and the privilege carbon context does not keep the authenticated user's name for these types of flows. That's why the username had to resolved. 
https://github.com/wso2-extensions/identity-organization-management-core/pull/10
